src/store/modules/models/helpers.js

```diff
@@ -195,5 +196,39 @@ const helpers = {
         return state.library[type];
     }
   },
+  replaceIdsUpdateInfoForCloning(story, idMap) {
+    const clonedStory = _.cloneDeep(story);
+    clonedStory.doors.forEach((door) => {
+      const newID = idFactory.generate();
+      idMap[door.id] = newID;
+      door.id = newID;
+    });
+    clonedStory.windows.forEach((window) => {
+      const newID = idFactory.generate();
+      idMap[window.id] = newID;
+      window.id = newID;
+    });
+    clonedStory.spaces.forEach((space) => {
+      const newID = idFactory.generate();
+      idMap[space.id] = newID;
+      space.id = newID;
+      space.name = '';
+      space.face_id = idMap[space.face_id];
+    });
+    clonedStory.shading.forEach((shade) => {
+      const newID = idFactory.generate();
+      idMap[shade.id] = newID;
+      shade.id = newID;
+      shade.name = '';
+      shade.face_id = idMap[shade.face_id];
+    });
+    return { clonedStory };
+  },
+
+  replaceSpaceNamesForCloning(state, type, story) {
+    story.spaces.forEach((space) => {
```

Comment:
I'd suggest replacing this with `story.spaces = story.spaces.map((space) ...`.  On large plans, this is really slow because vue's setter keeps getting triggered.

Reply:
Implemented. I also used map for `replaceIdsUpdateInfoForCloning` as well.